<a href="https://colab.research.google.com/github/lwehausen/Python/blob/main/Cubs_Data_Analysis_Data_Viz.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

This file displays data analysis and data visualizations in Python.

The cubs csv file contains Cubs data between 1876-2022. The variables included are: Year, Team (Tm), League (Lg), Games Played (G), Wins (W), Losses (L), Ties, Winning percentage (W-L%) , Pythagorean winning percentage (pythW-L%), Finish, Number of Fair Ground Balls Hit (GB), Playoffs, Runs (R), Runs Allowed (RA), Attendance, BatAge, PAge, #Bat, #P, Top Player, Managers

Using the cubs csv file, answer the following questions:

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
cubs = pd.read_csv("https://raw.githubusercontent.com/gjm112/DSCI401/main/data/cubs_all_time.csv")

How many total games have the Cubs won between 1876
and 2022?

In [2]:
sum(cubs.W)

11141

The Cubs have won 11141 games between 1876-2022.

How many total games have the Cubs lost between 1876
and 2022?

In [3]:
sum(cubs.L)

10592

The Cubs have lost 10592 games between 1876 and 2022.

What year did the Cubs score the most runs?

In [ ]:
max(cubs.R)
np.argmax(cubs.W)
cubs.Year[np.argmax(cubs.R)]

1894

What year did the
Cubs score the fewest runs?

In [ ]:
min(cubs.R)
np.argmin(cubs.W)
cubs.Year[np.argmin(cubs.R)]

2020

Note that the Cubs had their worst year in 2020, the same year that the COVID-19 pandemic started. Despite their lack of runs, no players had positive test results during the season!

In how many seasons was the Cubs total attendance (i.e. the variable
Attendance) over 3 million?

In [ ]:
cubs[cubs['Attendance'] >= 3000000]
sum(cubs.Attendance > 3000000)

12

There were 12 seasons where the attendance was greater than or equal to 3 million.